clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala

```diff
@@ -49,6 +50,18 @@ object UncommittedGarbageCollector {
     )
     // Read objects from data path (new repository structure)
     var dataDF = new ParallelDataLister().listData(configMapper, dataPath)
+
+    // Save first slice from data prefix
+    if (!dataDF.isEmpty) {
+      // Need the before filter to to exclude slices that are not actually read
+      val slices = dataDF.filter(
+        col("last_modified") < before.getTime && !col("base_address").startsWith(repo)
+      )
```

Comment:
Why does it mean we haven't read it?

Reply:
We need to choose the first slice that we actually read data from. When we list the data - we provide a buffer time (before) which we filter all objects which have a creation date < before. If we select the first slice without this filter we might choose a slice that its data wasn't read at all. If we use this slice in the incremental run it will mean that we will stop reading data too early and miss objects that weren't read in the previous run